### ISSUE TITLE
[Authentication code] Modal missing attributes to convey context

### ISSUE THREAD

**Issue by svinkle** [CovidShield/portal#17]
## Issue summary

Each modal control and modal window are missing attributes to convey context. These attributes are used to indicate the presence of a "modal" element which assistive technology users require to help describe the purpose of the current context.

<img width="471" alt="Screen Shot 2020-05-18 at 11 04 37 AM" src="https://user-images.githubusercontent.com/1392632/82231897-d79faa80-98fb-11ea-96ee-932ab6fd4fcb.png">

#### HTML

```html
<button aria-label="More information" class="help__activator" type="button" …>
  <!-- … -->
</button>
<!-- … -->
<div class="modal" aria-labelledby="modal-title" role="dialog" tabindex="-1">
  <!-- … -->
</div>

<!-- … -->
```

## Steps to reproduce

1. Go to https://portal-rails.myshopify.io/ with with any supported browser
2. Login with `admin`/`password`
3. With an active screen reader, <kbd>Tab</kbd> to each modal control

### Expected

- Modal components to feature accompanying attributes to convey context.

### Actual

- Some modal component attributes are missing.

## Recommendation

1. For each modal control, add the [`aria-haspopup` attribute](https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup), setting its value to `dialog`.
2. For each modal window, add the [`aria-modal` attribute](https://www.w3.org/TR/wai-aria-1.1/#aria-modal), setting its value to `true`.

#### HTML

```html
<button
  aria-label="More information"
  class="help__activator"
  type="button"
  aria-haspopup="dialog"
  …
>
  <!-- … -->
</button>
<!-- … -->
<div
  class="modal"
  aria-labelledby="modal-title"
  role="dialog"
  aria-modal="true"
  tabindex="-1"
>
  <!-- … -->
</div>

<!-- … -->
```

## Specifications

- **Component:** Authentication code
- **WCAG Principle:** Robust
- **WCAG SC:** 4.1.2 Name, Role, Value
- **Severity:** High
- **Effort:** Low
